sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,99 @@ def is_consistent(self):
+        >>> omega = time**(-1)
+        >>> dimsys_SI.get_dimensionless_numbers(resistance, inductance, capacitance, voltage_amp, current_amp, omega)
+        [Dimension(capacitance*resistance**2/inductance), Dimension(current_amp*resistance/voltage_amp), Dimension(inductance*omega/resistance)]
+
+        The first dimensionless quantity `\zeta^2 = \frac{C R^2}{L}` shows the quantity `\zeta` is known as the damping factor of the
+        RLC circuit, and the last dimensionless quantity `\frac{L*\omega}{R}` is reminiscent of the definition of the attenuation.
+        The second dimensionless quantity is something that simply comes from Ohm's law.
+
+        References
+        ==========
+
+        - https://en.wikipedia.org/wiki/Buckingham_%CF%80_theorem
+        - https://en.wikipedia.org/wiki/RLC_circuit
+        """
+        number_of_quantities = len(list_of_derived_quantities)
+        exponent_matrix = []
+        # We extract the exponent matrix from the dimensional dependencies of the quantities
+        for m in range(number_of_quantities):
+            element_for_quantity = [0 for k in range(len(self.base_dims))]
+            dimensions = self.get_dimensional_dependencies(list_of_derived_quantities[m])
+            for base_dims in dimensions:
+                element_for_quantity[self.base_dims.index(base_dims)] = dimensions[base_dims]
+            exponent_matrix.append(element_for_quantity)
+        # Next we obtain the null space of the exponent matrix
+        exponent_matrix = Matrix(exponent_matrix)
+        return exponent_matrix.T.nullspace(simplify=True)
+
+    def verify_dimensionless_numbers(self, *list_of_derived_quantities):
+        """
+        Give the objects of class Dimension that represent the dimensionless quantities.
+        To be preferably used only for test purposes, please refer to /units/test/test_dimensionless.py
+        """
+        exponents = self.buckingham_pi_theorem(*list_of_derived_quantities)
+        null_space_dims = len(exponents)
+        number_of_quantities = len(list_of_derived_quantities)
+        set_of_dimless_nums = [Dimension(1) for k in range(null_space_dims)]
+        # Here we take the dot product of the nullspace with the derived quantities
+        for k in range(null_space_dims):
+            for m in range(number_of_quantities):
+                set_of_dimless_nums[k] = set_of_dimless_nums[k]*(list_of_derived_quantities[m]**exponents[k][m])
+        return set_of_dimless_nums
+
+    def get_dimensionless_numbers(self, *list_of_derived_quantities):
+        """
+        Give the resultant list the quantities (as an instance of class Dimension), which are actually the dimensionless numbers
+        in the original unit system.
+        """
+        exponents = self.buckingham_pi_theorem(*list_of_derived_quantities)
+        import traceback, re
+        args_storer = []
+        for elems in traceback.extract_stack(): args_storer.append(elems.line)
```

Comment:
What is this code doing? Surely there is a way to do this that doesn't involve parsing traceback strings.

Reply:
So my intention with the Buckingham Pi API is that the user may be able to input his set of dimensions as named variables (or even simply a list of dimensionful quantities as simple instances of class Dimension) and the possible forms of output are two in number:

1. Either named variables are passed (by that I mean something like force = Dimension('mass*length/time**2'), density = Dimension('mass/length**3'), diffusivity = Dimension('length**2/time')  and so on) and then in the output the user gets to see the actual 'dimensionless quantities' as power law combinations that comes from the buckingham pi procedure (for example, [Dimension('force/density/diffusivity**2'), ..., ..., ...]). To recall those names, I wanted the addresses and the variable names passed as a parameter to the function live in scope so that's why I used strings from the traceback.stack(). I also tried using packages like inspect, symtable and ast etc. but not much to avail. But in the hindsight I feel this approach might ain't work in case of a user merely passing a list of Dimension objects without declaring a new variable. But still, I am welcome to suggestions in this option.

2. Just return the null-space of the exponent matrix and explain in the documentation to the user as to how to interpret the output with examples. This sounds more traditional and feasible at the time being. Here I wanted to ask you a question: do IDEs like Anaconda, Jupyter have Sphinx by default? I had Anaconda quite a while ago but since I have a limited memory Mac and I received an FEM project so I had to compromise my Anaconda. The question arises because my terminal does not process make html command from sympy/docs and is unable to render the LaTeX part of the documentation and hence unable to decipher problems with CircleCI. Can you please help me with this? Thanks a lot in advance :)

Reply:
I'm still not completely following what you are trying to do (and you need to fix the formatting in your comment). However, in general, you should avoid functions that take in expressions as strings. Expressions should be constructed directly from the objects. 

We have a guide for building the documentation here https://docs.sympy.org/dev/guides/contributing/build-docs.html.

Reply:
HI Aaron ... Thanks a lot for the suggestions and the help with the docs. I have removed the part with the strings completely and retained the original form of the definitions with appropriate description in the documentation part of the code. I think Sphinx works for me now, and helped me quite a lot with the results of the LaTeX typesetting. Would like to know your reviews on the same :)